src/utils/middleware.ts

```diff
@@ -95,48 +124,61 @@ export const getSpidStrategyOptionsUpdater = (
               {
                 [serviceProviderConfig.spidTestEnvUrl]: "xx_testenv2"
               }
+            ).fold(
+              () => ({}),
+              _ => _
             )
           ]
         : []
     );
   return array
-    .sequence(taskEither)(idpOptionsTasks)
+    .sequence(task)(idpOptionsTasks)
     .map(idpOptionsRecords =>
       idpOptionsRecords.reduce((prev, current) => ({ ...prev, ...current }), {})
     )
     .map(idpOptionsRecord => {
       logSamlCertExpiration(serviceProviderConfig.publicCert);
-      return {
-        idp: idpOptionsRecord,
-        sp: {
-          ...samlConfig,
-          attributes: {
-            attributes: serviceProviderConfig.requiredAttributes,
-            name: "Required attributes"
-          },
-          identifierFormat:
-            "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
-          organization: serviceProviderConfig.organization,
-          signatureAlgorithm: "sha256"
-        }
-      };
+      return bindSpidStrategyOptions(
+        samlConfig,
+        serviceProviderConfig,
+        idpOptionsRecord
+      );
     });
 };
 
 const SPID_STRATEGY_OPTIONS_KEY = "spidStrategyOptions";
 
 /**
- * SPID strategy calls getSamlOptions() for every
- * SAML request. It extracts the options from a
- * shared variable set into the express app.
+ * upsertSpidStrategyOption() is called to set or update Spid Strategy Options.
```

Comment:
```suggestion
 * This method is called to set or update Spid Strategy Options.
```

(in case of rectorings on method namings)

Reply:
Fixed in a4bdb13